### ISSUE TITLE
Adding multiple Custom Icon in Folium

### ISSUE THREAD

**Issue by shubham039** [python-visualization/folium#744]
#### I am creating a Custom Icon in folium, and displaying it on a map, but when i try to add another coordinate with the same Icon, it fails.

```
import folium
map_osm = folium.Map(location=[45.3288, -121.6625])
icon_url = 'https://cdn1.iconfinder.com/data/icons/maps-locations-2/96/Geo2-Number-512.png'
icon = folium.features.CustomIcon(icon_url,icon_size=(28, 30))  # Creating a custom Icon
folium.Marker(location=[45.3288, -121.6625],icon=icon).add_to(map_osm)  #adding it to the map
map_osm
```
![image](https://user-images.githubusercontent.com/32260408/31481725-784fd852-af42-11e7-967f-28d644e3526a.png)

#### Now adding another location with the same icon
```
folium.Marker(location=[46.3288, -122.6625],icon=icon).add_to(map_osm)
map_osm
```
![image](https://user-images.githubusercontent.com/32260408/31481794-dc4572d6-af42-11e7-96c3-cc8253e78d06.png)

[The map_osm points to new location with its default icon, and the previous pin is removed.]

**Comment by shubham039** [comment]
I got this solved, it needs icon to be initialised everytime you add a marker.
```
icon = folium.features.CustomIcon(icon_url,icon_size=(28, 30))
folium.Marker(location=[46.3288, -122.6625],icon=icon).add_to(map_osm)
map_osm
```

**Comment by ninabel** [comment]
When icon have to be initialised everytime we got huge file. It's a bug.

**Comment by Conengmo** [comment]
I guess you're right @ninabel, do you want to look into this and open a PR? I'm thinking now we should let an element have multiple parents in a way, but render it only once. Or something.

**Comment by hbarovertwo** [comment]
Was this ever addressed in another issue? Because I notice the bug mentioned above by @ninabel still persists.

**Comment by hbshrestha** [comment]
The bug is still persisting. It has not been solved yet. @hbarovertwo

**Comment by lowdowner** [comment]
I managed to solved this problem by initiliasing the custom icon within a for loop.
Example below.

```
   world = folium.Map(
    zoom_start=2
    )

    x = data[['Lat', 'Lon', 'Color']].copy()

    for index, row in x.iterrows():
         pushpin = folium.features.CustomIcon('/content/drive/My Drive/Colab Notebooks/pushpin.png', icon_size=(30,30))
         folium.Marker([row['Lat'], row['Lon']],
                                    icon=pushpin,
                                    popup=row['Color'],
                                   ).add_to(world)


    world
```

**Comment by Playelectro** [comment]
Is this issue fix or is there a good work-around that does not load redundant copies of the same image, thus slowing the server?
